### ISSUE TITLE
UTF-8 Queries not properly encoded (and this kills the process)

### ISSUE THREAD

**Issue by andyleclair** [pepsico-ecommerce/snowflex#70]
Hi!
I'm making some queries with Snowflex and noting this in my Snowflake query console: 

![image](https://user-images.githubusercontent.com/109716/227629372-514a7b59-4115-43f4-9621-7e4b06a9dd3f.png)

The query I'm making includes the event name: `[OBD-2] Finalizou tutorial do prontuário`

I am guessing that there's a missing UTF-8 conversion in the query somewhere?

I have also noticed that making certain queries (try doing one with the first value from [this document ](https://community.snowflake.com/s/article/Does-the-Snowflake-ODBC-Driver-support-non-Latin-characters)) causes the connection to drop entirely and not be recreated

![image](https://user-images.githubusercontent.com/109716/227629758-672eeda3-e3d8-4dd0-bb0d-31a604aa8452.png)

From then on, any query will return `{:error, [:connection_closed, "No SQL-driver information available"]}`

**Comment by andyleclair** [comment]
We were, yes. After looking into it more, the Erlang ODBC docs specify that the query is a list of ASCII characters, so my guess is that's where the real issue lies. https://www.erlang.org/doc/man/odbc.html

I tried a few variations of sending queries to `:odbc` directly with varying attempts at character coercion from the `:unicode` module, but no dice there. Maybe I should raise a bug with the Erlang team?

I didn't get anywhere with the process not being restarted, once we discovered this limitation, using ODBC at all was a non-starter for us, as we accept user queries and we need it to work with UTF-8 data. We're using (Avalanche)[https://github.com/HGInsights/avalanche] now, and it's working well for us.

I'll close this, but probably something people should be aware of?
